lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -578,6 +645,12 @@ void SingleExitLoopTransformer::ensureSingleExitBlock() {
     }
     loop->addBasicBlockToLoop(outsideBlock, LI->getBase());
   }
+  if (cloner.hasCloned()) {
+    // TODO(https://bugs.swift.org/browse/SR-8336): the transformations here are
+    // simple that we should be able to incrementally update the DI & PDI.
+    DI->recalculate(*currentFn);
+    PDI->recalculate(*currentFn);
```

Comment:
FWIW, I think this approach is totally fine, assuming that this isn't commonly necessary.  If it is commonly necessary, it would be interesting to see what the patterns are and maybe a trivial local update would work.